### ISSUE TITLE
New structure proposal

### ISSUE THREAD

**Issue by mahnouman** [tamiat/tamiat#137]
Hello,
This issue is for creating a new files and folders structure, keeping in mind that in the CMS there are/should be mainly two  trees, one for admin and related components and the other is for the website itself.

We would assume that the user would be typically changing the website architecture but leaving the admin structure intact. 

One important thing to consider is to keep compatibility with vue-cli src webpack template. Means that we can easily init a project using our webpack template repo, which is forked from the official vue webpack template.

Another point to consider is the styles, not sure if the new structure would make any difference from the styles point of view, but the expectation is that the styles of the admin area should not affect the website styles and vice versa, and the mechanism for doing so should not rely on the user taking care not to change something in the app (Not admin) area.

So, this issue is about proposing a new structure not actually implementing it.

**Comment by Pajicc** [comment]
Basic structure:
![basicstructure](https://user-images.githubusercontent.com/18647471/36551517-0735390e-17f8-11e8-99bd-d361210085c1.png)
ps: In website content -> pages - we currently only have Home component

Admin structure:
![adminstructure](https://user-images.githubusercontent.com/18647471/36551516-070cb6dc-17f8-11e8-87f3-17c6fa71384a.png)

The idea is to keep the usual Vue.js structure in Admin as well as in Website content so that developers can get around it.

Let me know what you think

**Comment by kieranwild** [comment]
This looks great 👍

**Comment by mahnouman** [comment]
Hi Srdjan, ok, so can we habe website structure all in one place, for example, I see the assets here are combined for both admin and website. My idea here is that, if someone imported a website template, usually it makes sense to have it extracted in one place.

**Comment by Pajicc** [comment]
Then we can maybe define all css for admin also in Admin content (leave it like it is now, or make it external - create new Assets folder just for Admin)
So the current Assets would belong only to Website content, would that be okay?
We can also group all website content in Website folder if you want, but this is ok too i think

**Comment by mahnouman** [comment]
A visualized example would help. But we need to maintain the concept, admin and whatever inside admin is ours. Website is the user and we should organize things in a way that makes less frustration for the user. So we better to ask him/her to put his code under one hood, does not ask him to make things different than a usual vue project. We also need to consider plugins idea. A place where he can grab a third part code that is pluggable into the solution to do something for either the admin or website. We just need to care now for the structure. and we care that the styles are not getting conflicted or broken.

**Comment by BartoszBaczek** [comment]
@Pajicc, IMO there should be one folder for shared vue components.

**Comment by Pajicc** [comment]
@BartoszBaczek There is, in Admin content. We can add it too in website content but that's really on users choice. We shouldn't mix them up.
@mahnouman This structure I made provides all that you want i think, we'll just create the assets folder for both contents, so the styles are spereated too. I can create new visualized example

**Comment by BartoszBaczek** [comment]
@Pajicc Yup, I've seen it - my idea was, that maybe we should mix admin shared content with website content, as long as we create one bundle for both admin and client (as it is now I believe).

**Comment by mahnouman** [comment]
Hi @Pajicc , yes please a visual example will help. Thanks

**Comment by Pajicc** [comment]
![screenshot_1](https://user-images.githubusercontent.com/18647471/36675340-fbd34560-1b08-11e8-9211-056d5cc0bcdd.png)

**Comment by Pajicc** [comment]
Hm, I would suggest we keep the web content in src as it is, because it's the usual Vue.js structure. We can maybe have admin outside of src but this is okay also I think

**Comment by mahnouman** [comment]
sorry, it was not clear
I mean /src/admin and /src/web

**Comment by Pajicc** [comment]
Sure, it could work

**Comment by mahnouman** [comment]
Hi @Pajicc So, If I want to customize the website, everything I would need to do till be inside /src/web ? The idea here is that a developer download the CMS and work his way with customization from inside one directory.

**Comment by Pajicc** [comment]
Yes, all you need would be in web. But basic vue structure isn't wrapped in web folder, just in src. So I think for developers, it would be more understandable if we do it like in the last visual I posted

**Comment by mahnouman** [comment]
Yeah, but it will not serve the purpose then, if I delete everything inside the directory that contains the website code, the CMS would still function at least from /admin point of view. Thus the root cannot be /src.

**Comment by Pajicc** [comment]
With this structure we will seperate the functionality of web and admin content, so yea, if you delete web content, the CMS part would still work, at least from admin view. But what you're saying is that you don't want it to be like that?
By wrapping content in web folder we won't achieve anything differently.
We can maybe put admin content outside of src, so basically in the src would be only web content (as usual Vue structure)

**Comment by mahnouman** [comment]
You know the web template which we ship with the CMS, I want it to be very easy to replace. The way to make it work it to have everything about it inside one directory, so the user can replace it easily with something else. I initially wanted to call it /src/admin and /src/app . if /src/app is not possible we can name it /src/web , Let us adjust the code to be have the /src/admin and /src/app and take it from there

**Comment by Pajicc** [comment]
Yea, okay. One thing that could cause a problem would be the routes, because we use the same one for both admin and web, but that could be discussed in other issue (#138)

**Comment by mahnouman** [comment]
Ideally we would leave the routing on /src , but yes we would do this in the routing issue. You can leave the routing on /src for now
